python/ambassador/ir/irtracing.py

```diff
@@ -79,6 +79,10 @@ def setup(self, ir: 'IR', aconf: Config) -> bool:
                 if not driver_config['collector_endpoint_version'] in ['HTTP_JSON_V1', 'HTTP_JSON', 'HTTP_PROTO']:
                     self.post_error(RichStatus.fromError("collector_endpoint_version must be one of 'HTTP_JSON_V1, HTTP_JSON, HTTP_PROTO'"))
                     return False
+            if 'propagation_modes' in driver_config:
+                if not all(item in ['ENVOY', 'LIGHTSTEP', 'B3', 'TRACE_CONTEXT'] for item in driver_config['propagation_modes']):
+                    self.post_error(RichStatus.fromError("propagation_modes must contain any of 'ENVOY, LIGHTSTEP, B3, TRACE_CONTEXT'"))
+                    return False
```

Comment:
It shouldn't be necessary to check this here; the schema-based validator will have already rejected the resource if it doesn't match what's in the `+kubebuilder:validation:Enum=` comment.

(This is only true since 2.1.2, which is why there's still a bunch of now-superflous code that does these type of checks.)

Reply:
Nice!